2022-04-08 09:43:46,420 INFO  ==============================================================
2022-04-08 09:43:46,421 INFO  Id:    monitoring_with_simulated_messages
2022-04-08 09:43:46,422 INFO  Dir:   monitoring/
2022-04-08 09:43:46,424 INFO  Title: Validate the tedge dm agent messages that are produced from the simulated collectd messages
2022-04-08 09:43:46,425 DEBUG Execution order hint: 0.0
2022-04-08 09:43:46,426 INFO  ==============================================================
2022-04-08 09:43:46,437 DEBUG Process parameters for sudo<collectd>
  command line : /usr/bin/sudo systemctl stop collectd
    arg #1     : systemctl
    arg #2     : stop
    arg #3     : collectd
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:46,445 DEBUG Waiting up to 600 secs for process sudo<collectd> (pid 24153)
2022-04-08 09:43:47,054 INFO  Executed sudo<collectd>, exit status 0
2022-04-08 09:43:47,056 DEBUG Process parameters for sudo<collectd_mapper>
  command line : /usr/bin/sudo systemctl start tedge-mapper-collectd
    arg #1     : systemctl
    arg #2     : start
    arg #3     : tedge-mapper-collectd
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd_mapper.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd_mapper.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:47,064 DEBUG Waiting up to 600 secs for process sudo<collectd_mapper> (pid 24155)
2022-04-08 09:43:47,649 INFO  Executed sudo<collectd_mapper>, exit status 0
2022-04-08 09:43:47,650 DEBUG --- test execute
2022-04-08 09:43:47,653 DEBUG Process parameters for sudo<tedge_sub>
  command line : /usr/bin/sudo /usr/bin/tedge mqtt sub --no-topic 'tedge/#'
    arg #1     : /usr/bin/tedge
    arg #2     : mqtt
    arg #3     : sub
    arg #4     : --no-topic
    arg #5     : tedge/#
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_sub.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_sub.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:47,661 INFO  Started sudo<tedge_sub> with process id 24162
2022-04-08 09:43:47,767 DEBUG Process parameters for sudo<tedge_temp>
  command line : /usr/bin/sudo /usr/bin/tedge mqtt pub collectd/host/temperature/temp 1649411027.7645965:25.5
    arg #1     : /usr/bin/tedge
    arg #2     : mqtt
    arg #3     : pub
    arg #4     : collectd/host/temperature/temp
    arg #5     : 1649411027.7645965:25.5
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_temp.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_temp.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:47,775 DEBUG Waiting up to 600 secs for process sudo<tedge_temp> (pid 24163)
2022-04-08 09:43:48,332 INFO  Executed sudo<tedge_temp>, exit status 0
2022-04-08 09:43:48,335 DEBUG Process parameters for sudo<tedge_pres>
  command line : /usr/bin/sudo /usr/bin/tedge mqtt pub collectd/host/pressure/pres 1649411028.3339937:500.5
    arg #1     : /usr/bin/tedge
    arg #2     : mqtt
    arg #3     : pub
    arg #4     : collectd/host/pressure/pres
    arg #5     : 1649411028.3339937:500.5
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_pres.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/tedge_pres.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:48,343 DEBUG Waiting up to 600 secs for process sudo<tedge_pres> (pid 24166)
2022-04-08 09:43:48,862 INFO  Executed sudo<tedge_pres>, exit status 0
2022-04-08 09:43:49,865 DEBUG Process parameters for sudo<kill_out>
  command line : /usr/bin/sudo killall tedge
    arg #1     : killall
    arg #2     : tedge
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/kill_out.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/kill_out.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:49,873 DEBUG Waiting up to 600 secs for process sudo<kill_out> (pid 24168)
2022-04-08 09:43:50,390 INFO  Executed sudo<kill_out>, exit status 0
2022-04-08 09:43:50,391 DEBUG --- test validate
2022-04-08 09:43:50,394 INFO  0

2022-04-08 09:43:50,395 WARN  TypeError occurred while running test - 'int' object is not subscriptable
Traceback (most recent call last):
  File "/home/pi/env-pysys/lib/python3.7/site-packages/pysys/baserunner.py", line 1296, in __call__
    self.testObj.validate()
  File "/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/run.py", line 98, in validate
    collectd_msg_validation_result=self.validate_json(),
  File "/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/run.py", line 108, in validate_json
    if not self.validate_time():
  File "/home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/run.py", line 127, in validate_time
    if self.js_msg["time"]:
TypeError: 'int' object is not subscriptable
2022-04-08 09:43:50,410 WARN  TypeError: 'int' object is not subscriptable ... blocked 
2022-04-08 09:43:50,412 DEBUG --- test cleanup
2022-04-08 09:43:50,413 INFO  
2022-04-08 09:43:50,413 INFO  cleanup:
2022-04-08 09:43:50,414 DEBUG Running registered cleanup function: <bound method MonitoringWithSimulatedMessages.monitoring_cleanup of <MonitoringWithSimulatedMessages object at 0x75d328b0>>
2022-04-08 09:43:50,415 INFO  monitoring_cleanup
2022-04-08 09:43:50,417 DEBUG Process parameters for sudo<collectd_mapper>
  command line : /usr/bin/sudo systemctl stop tedge-mapper-collectd
    arg #1     : systemctl
    arg #2     : stop
    arg #3     : tedge-mapper-collectd
  working dir  : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux
  stdout       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd_mapper.out
  stderr       : /home/pi/actions-runner/_work/thin-edge.io/thin-edge.io/tests/PySys/monitoring/monitoring_with_simulated_messages/Output/linux/collectd_mapper.err
  environment  : LD_LIBRARY_PATH=/usr/lib
  environment  : PATH=/bin:/usr/bin:/usr/sbin:/usr/local/bin
                   #1  /bin
                   #2  /usr/bin
                   #3  /usr/sbin
                   #4  /usr/local/bin
2022-04-08 09:43:50,425 DEBUG Waiting up to 600 secs for process sudo<collectd_mapper> (pid 24170)
2022-04-08 09:43:51,233 INFO  Executed sudo<collectd_mapper>, exit status 0
2022-04-08 09:43:51,234 DEBUG ProcessUser cleanup function done.
2022-04-08 09:43:51,238 INFO  
2022-04-08 09:43:51,238 INFO  Test duration: 4.83 secs
2022-04-08 09:43:51,239 INFO  Test final outcome:  BLOCKED
2022-04-08 09:43:51,240 INFO  Test outcome reason: TypeError: 'int' object is not subscriptable
2022-04-08 09:43:51,241 INFO  
